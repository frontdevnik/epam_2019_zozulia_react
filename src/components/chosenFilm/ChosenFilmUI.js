import React from 'react';
import RatingStart from '../movies/ratingStart/RatingStart';
import withTranslation from '../../hocs/withTranslation';
import style from './chosenFilm.module.scss';

function ChosenFilmUI(props) {
  const { title, posterUrl, stars, likes, genres, director, description, id } = props.movie;
  const { actors, editMovie, removeMovie, choosenFilm_likes, choosenFilm_edit, choosenFilm_delete, choosenFilm_director, choosenFilm_actors, choosenFilm_genres, choosenFilm_description } = props;

  return (
    <>
      <div className={style.shortInfo}>
        <h3>{title}</h3>
        <p>
          {choosenFilm_likes}:
          {likes}
        </p>
        <RatingStart stars={stars} id={+id} />
        <div className={style.buttonContainer}>
          <button onClick={editMovie} className={style.changeButton}>{choosenFilm_edit}</button>
          <button onClick={removeMovie(id)} className={style.changeButton}>{choosenFilm_delete}</button>
        </div>
      </div>
      <article className={style.fullInfo}>
        <img className={style.mainImg} src={posterUrl} alt={title} />
        <p>
          <strong>{choosenFilm_director}: </strong>
          {director}
        </p>
        <p>
          <strong>{choosenFilm_actors}: </strong>
          {actors}
        </p>
        <p>
          <strong>{choosenFilm_genres}: </strong>
          {genres}
        </p>
        <p>
          <strong>{choosenFilm_description}: </strong>
          {description}
        </p>
      </article>
    </>
  )
};

const withTranslationWords = withTranslation(['choosenFilm_likes', 'choosenFilm_edit', 'choosenFilm_delete', 'choosenFilm_director', 'choosenFilm_actors', 'choosenFilm_genres', 'choosenFilm_description']);

export default withTranslationWords(ChosenFilmUI);