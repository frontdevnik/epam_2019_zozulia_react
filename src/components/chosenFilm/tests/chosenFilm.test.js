import React from "react";
import { shallow } from "enzyme";
import ChosenFilmUI from "../ChosenFilmUI";
import styles from "./chosenFilm.module.scss";

describe("<ChosenFilmUI />", () => {
  let wrapper;

  const props = {
    editMovie: jest.fn(),
    removeMovieById: jest.fn(),
    removeMovie: jest.fn(() => props.removeMovieById),
    choosenFilm_edit: "Edit",
    choosenFilm_delete: "Delete",
    actors: [],
    choosenFilm_likes: "Likes",
    choosenFilm_director: "Director",
    choosenFilm_actors: "Actors",
    choosenFilm_genres: "Genres",
    choosenFilm_description: "Description",
    movie: {
      title: "Title",
      posterUrl: "posterUrl",
      stars: "4",
      likes: "3",
      genres: ["Genres"],
      director: "Director",
      description: "Description",
      id: 1,
    },
  };

  beforeEach(() => {
    wrapper = shallow(<ChosenFilmUI {...props} />).dive();
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should invoke editMovie after button click", () => {
    const button = wrapper.find("button").at(0);
    button.simulate("click");
    expect(props.editMovie).toBeCalledTimes(1);
    expect(button.text()).toEqual(props.choosenFilm_edit);
  });

  it("should invoke removeMovie after button click", () => {
    const button = wrapper.find("button").at(1);
    button.simulate("click");
    expect(props.removeMovie()).toBeCalledTimes(1);
    expect(button.text()).toEqual(props.choosenFilm_delete);
  });
});
