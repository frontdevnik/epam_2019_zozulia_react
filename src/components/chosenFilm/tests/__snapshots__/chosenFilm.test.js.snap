// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChosenFilmUI /> should render correctly 1`] = `
"<Fragment>
  <div className=\\"shortInfo\\">
    <h3>
      Title
    </h3>
    <p>
      Likes
      :
      3
    </p>
    <RatingStart stars=\\"4\\" id={1} />
    <div className=\\"buttonContainer\\">
      <button onClick={[Function: mockConstructor]} className=\\"changeButton\\">
        Edit
      </button>
      <button onClick={[Function: mockConstructor]} className=\\"changeButton\\">
        Delete
      </button>
    </div>
  </div>
  <article className=\\"fullInfo\\">
    <img className=\\"mainImg\\" src=\\"posterUrl\\" alt=\\"Title\\" />
    <p>
      <strong>
        Director
        : 
      </strong>
      Director
    </p>
    <p>
      <strong>
        Actors
        : 
      </strong>
    </p>
    <p>
      <strong>
        Genres
        : 
      </strong>
      Genres
    </p>
    <p>
      <strong>
        Description
        : 
      </strong>
      Description
    </p>
  </article>
</Fragment>"
`;
